### DISCUSSION TITLE
Running hubs-cloud client locally

### DISCUSSION THREAD

**Discussion by jonathan2daniels** [mozilla/hubs#2713] [Hubs Cloud Questions] [answered]
Hello everyone, I'm new to hubs-cloud and I'm having a ton of fun playing around with it. I am also experimenting with a custom hubs-cloud client. However, I'm having trouble running it locally because of self signed certificate issues from browsers and sometimes it doesn't even load though it compiles successfully in my cli. This usually happens after I have deployed changes to my hubs-cloud project. Please help!

**Comment by FarhadG** [comment, marked as answer]
You can by pass this with allowing the browser to allow `localhost`. 

- In Chrome, you can try this: https://community.spiceworks.com/how_to/154601-chrome-allow-localhost-site-even-without-https-certificate
- For Firefox, you just have to click advanced on the blocked page and manually `allow` for the site

Good luck!

**Reply by robinkwilson** [reply]
@jonathan2daniels Did this work for you?

**Reply by jonathan2daniels** [reply]
Hello Farhad, I tried what you suggested and I don't get permission errors anymore. However, the site would still not load. 😣

**Comment by robinkwilson** [comment]
@jonathan2daniels 
**Hosting locally questions:**
Do you have any error logs in the hubs client console? 
Any failed requests in the Network tab? Any information would help!

**HC questions:**
Do you have database pausing on? Is the stack in Offline mode?
Are you basing the branch on github.com/hubs "hubs-cloud" branch and NOT "master" branch?
Does the newly deployed custom client load for you? Remember you can run `npm run undeploy` to remove the custom client from your HC instance.

Thanks! These will narrow the issue down!

**Reply by jonathan2daniels** [reply]
Hello Robin,

I have no errors in the client console and I am on the right branch - hubs-cloud. However, I do get an error in the browser that says my certificate is invalid. This happens in chrome. I have tried what @FarhadG has suggested and this didn't solve the problem.
![Screen Shot 2020-07-29 at 9 32 04 PM](https://user-images.githubusercontent.com/39608305/88801432-ab9a8280-d1e4-11ea-8209-56d7fe385029.jpeg)
![Screen Shot 2020-07-29 at 9 32 34 PM](https://user-images.githubusercontent.com/39608305/88801448-b05f3680-d1e4-11ea-98dd-87313e9736bf.jpeg)
![Screen Shot 2020-07-29 at 9 34 05 PM](https://user-images.githubusercontent.com/39608305/88801451-b0f7cd00-d1e4-11ea-9c7d-48d0e78bf2fa.jpeg)
![Screen Shot 2020-07-29 at 9 34 25 PM](https://user-images.githubusercontent.com/39608305/88801452-b1906380-d1e4-11ea-8f67-31fce3f620e7.jpeg)

**Comment by robinkwilson** [comment]
In Chrome, usually you click "Advanced" > "Accept" like the screenshots in https://kinsta.com/blog/your-connection-is-not-private/

Can you try running the client then going to [https://hubs.local:8080](https://hubs.local:8080)? For the hubs dev environment, we use this address instead of localhost.

If that doesn't show up, I updated my hosts file via [this guide for updating a Mac's host file](https://www.techjunkie.com/edit-hosts-file-mac-os-x/) to edit my hosts file.

My `hosts` file:
```
127.0.0.1	localhost
::1             localhost
127.0.0.1       hubs.local
127.0.0.1	hubs-proxy.local
```

Let me know if this helps!

**Reply by jonathan2daniels** [reply]
I tried your suggestions, unfortunately it's still not working for me. I'm sorry if I haven't been able to describe the problem clearly. one thing i noticed is that before deployment, everything works just fine. I only start getting these browser issues once i have deployed to hubs cloud.

**Comment by bipinparajuli** [comment]
Any solution I'm getting same error. Through I can run from firebox browser but Failed to load glTF model SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data. Can't create room
